packages/strapi-design-system/src/v2/SimpleMenu/SimpleMenu.js

```diff
@@ -118,25 +122,30 @@ export const SimpleMenu = ({
         setFocusItem(defaultItemIndexToFocus);
       }
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [label]);
 
+  const handleOpen = useCallbackRef(onOpen);
+
+  const handleClose = useCallbackRef(onClose);
+
   useEffect(() => {
     if (didMount?.current) {
-      if (visible && typeof onOpen === 'function') {
-        onOpen();
-      } else if (typeof onClose === 'function') {
-        onClose();
+      if (visible) {
+        handleOpen();
+      } else {
+        handleClose();
       }
     } else {
       didMount.current = true;
     }
-  }, [didMount, visible]);
+  }, [didMount, visible, handleOpen, handleClose]);
 
   /* in case `label` is a custom react component, we know it is going to be
       a child of the menu button.
   */
   useEffect(() => {
-    if (React.isValidElement(label) && focusedItemIndex == -1) {
+    if (React.isValidElement(label) && focusedItemIndex === -1) {
```

Comment:
wondering why it was written this way, we're sure `focusedItemIndex` is never a string?

Reply:
not sure, but yes it'll be a number 👍🏼